ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/tree/TomlArrayOfTables.kt

```diff
@@ -65,6 +66,56 @@ public class TomlArrayOfTables(
             (0..index).joinToString(".") { sectionsList[it] }
         }
     }
+
+    override fun TomlEmitter.writeHeader(headerKey: TomlKey, config: TomlConfig) {
+        startTableArrayHeader()
+
+        headerKey.write(emitter = this, config)
+
+        endTableArrayHeader()
+    }
+
+    override fun TomlEmitter.writeChildren(
+        headerKey: TomlKey,
+        children: List<TomlNode>,
+        config: TomlConfig,
+        multiline: Boolean
+    ) {
+        val last = children.lastIndex
+
+        children.forEachIndexed { i, child ->
+            if (child is TomlArrayOfTablesElement) {
+                if (parent !is TomlArrayOfTablesElement) {
+                    emitIndent()
+                }
+
+                writeHeader(headerKey, config)
+
+                if (!child.hasNoChildren()) {
+                    emitNewLine()
+                }
+
+                indent()
+
+                child.write(emitter = this, config, multiline)
+
+                dedent()
+
+                if (i < last) {
+                    emitNewLine()
+
+                    // Primitive pairs have a single newline after, except when a
+                    // table follows.
+                    if (child !is TomlKeyValuePrimitive ||
+                            children[i + 1] is TomlTable) {
+                        emitNewLine()
+                    }
+                }
+            } else {
+                child.write(emitter = this, config, multiline)
```

Comment:
do we need to emit a newline here?

Reply:
Otherwise we have no blank line between blocks, so the output would look like this:

```toml
[[fruits]]
    a = "apple"
    b = "qwerty"
[[fruits]]
    a = "banana"
    b = "qwerty"
[[fruits]]
    a = "plantain"
    b = "qwerty"
```

Or is that the desired output?